WalletWasabi/WabiSabi/Backend/Banning/Whitelist.cs

```diff
@@ -78,15 +87,22 @@ public void RemoveAllExpired(TimeSpan releaseTime)
 
 	public bool TryGet(OutPoint utxo, [NotNullWhen(true)] out Innocent? innocent)
 	{
-		return Innocents.TryGetValue(utxo, out innocent);
-	}
+		if (!Innocents.TryGetValue(utxo, out innocent))
+		{
+			return false;
+		}
 
-	public IEnumerable<Innocent> GetInnocents()
-	{
-		return Innocents.Values;
+		if (innocent.TimeSpent > WabiSabiConfig.ReleaseFromWhitelistAfter)
+		{
+			TryRelease(innocent.Outpoint);
+			Logger.LogInfo($"1 utxo was expired and removed from Whitelist.");
+			return false;
+		}
+
+		return true;
 	}
 
-	public static async Task<Whitelist> CreateAndLoadFromFileAsync(string whitelistFilePath, CancellationToken cancel)
+	public static async Task<Whitelist> CreateAndLoadFromFileAsync(string whitelistFilePath, WabiSabiConfig wabiSabiConfig, CancellationToken cancel)
```

Comment:
I don't think tho' that we need the whole config file.
For paramteter OK, but inside for property we should save only `ReleaseFromWhitelistAfter` as thats the only thing we use.

Reply:
There is one huge advantage to this. We can change the config on the fly, and the Whitelist will pick up the changes.
(no need to restart the backend)

Reply:
Exactly. We need WabiSabiConfig to get the new value in case of change.

Reply:
Ahh okay, make sense.

Reply:
Yes. The magic happens in the `ConfigBase`, not in the `Whitelist`.

And since we directly referencing the `WabiSabiConfig` here

```
var allInnocentsToRemove = Innocents.Values.Where(innocent => innocent.TimeSpent > WabiSabiConfig.ReleaseFromWhitelistAfter);
```

The `WabiSabiConfig.ReleaseFromWhitelistAfter` will have the updated value.